provider/internal/java/dependency.go

```diff
@@ -97,52 +96,65 @@ func getMavenLocalRepoPath(mvnSettingsFile string) string {
 	return string(outb.String())
 }
 
-func (p *javaServiceClient) GetDependencyFallback(ctx context.Context) (map[uri.URI][]*provider.Dep, error) {
-	pomDependencyQuery := "//dependencies/dependency/*"
-	path := p.findPom()
-	file := uri.File(path)
-
-	absPath, err := filepath.Abs(path)
-	if err != nil {
-		return nil, err
-	}
+func (p *javaServiceClient) GetDependenciesFallback(ctx context.Context, location string) (map[uri.URI][]*provider.Dep, error) {
+	deps := []*provider.Dep{}
 
-	f, err := os.Open(absPath)
-	if err != nil {
-		return nil, err
-	}
-	doc, err := xmlquery.Parse(f)
-	if err != nil {
-		return nil, err
+	path := p.findPom()
+	if location != "" {
+		path = location
 	}
-	list, err := xmlquery.QueryAll(doc, pomDependencyQuery)
+	pom, err := gopom.Parse(path)
 	if err != nil {
 		return nil, err
 	}
-	deps := []*provider.Dep{}
-	dep := &provider.Dep{}
-	// TODO this is comedically janky
-	for _, node := range list {
-		if node.Data == "groupId" {
-			if dep.Name != "" {
-				deps = append(deps, dep)
-				dep = &provider.Dep{}
+
+	// have to get both <dependencies> and <dependencyManagement> dependencies (if present)
+	var pomDeps []gopom.Dependency
+	if pom.Dependencies != nil {
+		pomDeps = append(pomDeps, *pom.Dependencies...)
+	}
+	if pom.DependencyManagement != nil {
+		pomDeps = append(pomDeps, *pom.DependencyManagement.Dependencies...)
+	}
+
+	// add each dependency found
+	for _, d := range pomDeps {
+		dep := provider.Dep{}
+		dep.Name = fmt.Sprintf("%s.%s", *d.GroupID, *d.ArtifactID)
+		if *d.Version != "" {
+			if strings.Contains(*d.Version, "$") {
+				version := strings.TrimSuffix(strings.TrimPrefix(*d.Version, "${"), "}")
+				version = pom.Properties.Entries[version]
+				if version != "" {
+					dep.Version = version
+				}
+			} else {
+				dep.Version = *d.Version
 			}
-			dep.Name = node.InnerText()
-		} else if node.Data == "artifactId" {
-			dep.Name += "." + node.InnerText()
```

Comment:
Not sure about why this is concatenating with `.` instead of `:`?

Reply:
goal is to make dep name like, `com.foo.bar.artificatID`